### PR TITLE
feat(SSC): Use go.mod files instead of go.sum files

### PR DESCRIPTION
**Pull request by mmcqd**
[`go.sum` files are not lockfiles](https://stackoverflow.com/questions/69825533/why-does-go-sum-include-so-many-older-packages), and they contain multiple copies of versions of packages that are not actually being used by a project. This PR switches to using `go.mod` files as a source of dependency information for go projects. `go.mod` files actually _do_ function as lockfiles, as the go package manager automatically edits them to include only exact versions and all transitive dependencies.

PR checklist:

- [ ] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [ ] Tests included or PR comment includes a reproducible test plan
- [ ] Documentation is up-to-date
- [ ] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [ ] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Comment by aryx** [comment]
Note that for semgrep-pro @nmote also needed to parse go.mod files to resolve correctly names. We made a very
shallow parser for those go.mod files, but having all our parsers in OCaml would improve things a little bit here and avoid
code duplication.

**Comment by nmote** [comment]
Yeah, it definitely would be nice to have things consolidated. But, in this case the "parser" I wrote is dead simple and only a few lines long. I don't think there's a huge benefit to be had in this specific case by avoiding duplication.

File: cli/tests/performance/test_semdep.py [hunk 1]
```diff
@@ -14,7 +14,6 @@
         for file_size, max_time in [("10k", 3), ("50k", 15), ("100k", 30)]
         for target in [
             "Gemfile.lock",
-            "go.sum",
```
Comment (brendongo) [inline cli/tests/performance/test_semdep.py:17]: Do we want to add go.mod to the perf tests?

**Review by brendongo — APPROVED** [review]

File: cli/tests/performance/test_semdep.py (hunk 1 above)
Comment (mmcqd) [reply cli/tests/performance/test_semdep.py:17]: done

**Pull request merged**